apps/server/src/modules/task/uc/task.uc.ts

```diff
@@ -214,21 +216,38 @@ export class TaskUC {
 		return oneWeekAgo;
 	}
 
-	async delete(userId: EntityId, taskId: EntityId) {
+	async create(userId: EntityId, params: ITaskCreate): Promise<TaskWithStatusVo> {
+		this.checkNewTaskEnabled();
+
 		const user = await this.authorizationService.getUserWithPermissions(userId);
-		const task = await this.taskRepo.findById(taskId);
+		const course = await this.courseRepo.findById(params.courseId);
 
-		this.authorizationService.checkPermission(user, task, PermissionContextBuilder.write([]));
+		// TODO is this permissiosn checking parent correctly?
```

Comment:
"permissiosn"
also: not sure if the course should be the checking instance, didn't we talked in the past about the need to check with the user as the only suitable authorization place?

Reply:
edit: or at least only check the course, if there is a course given I think

Reply:
will check, that's a good TODO  :)

Reply:
is because the user wants to save a task inside a course, so he has to have permissions to write in the course.
In addition, he has to have permission to create a task (HOMEWORK_EDIT). We do the same on [task copy](https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/task/uc/task-copy.uc.ts#L89)